imports/client/components/PuzzlePage.tsx

```diff
@@ -460,8 +457,65 @@ class PuzzlePageMetadata extends React.Component<PuzzlePageMetadataProps> {
           ))
         }
       </span>
-    ) : null;
-    const numGuesses = this.props.guesses.length;
+    );
+
+    const puzzleLink = this.props.puzzle.url && (
+      <a
+        className="puzzle-metadata-external-link-button"
+        href={this.props.puzzle.url}
+        target="_blank"
+        rel="noreferrer noopener"
+      >
+        <FontAwesomeIcon fixedWidth icon={faPuzzlePiece} />
+        {' '}
+        <span className="link-label">Puzzle</span>
+      </a>
+    );
```

Comment:
Just for cleanliness: for these conditionally-generated elements, let's either use the ternary operator `(condition ? <element /> : null )` to make the value explicitly `null` or use `!!` to force the value to be `false` rather than accidentally some other false-y value that might get rendered differently.  (I've accidentally used the falsiness of 0 and wound up rendering the string 0 when I meant to render nothing.)